src/main/java/io/github/gitbucket/markedj/extension/Extension.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2023 GitBucket.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.gitbucket.markedj.extension;
+
+import io.github.gitbucket.markedj.Lexer;
+import io.github.gitbucket.markedj.Parser;
+import io.github.gitbucket.markedj.rule.Rule;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ *
+ * @author t.marx
+ */
+public interface Extension {
+	
+	public String getName();
+	
+	public Map<String, Rule> enhanceRules(Map<String, Rule> existingRules);
```

Comment:
Is adding rules actually necessary although `Extension` is called directly in `Parser` and `Lexer`?

Reply:
You are right, this is not needed anymore. code is removed.